How to fall in love 
1. Dig a hole
2. Name it love
3. Fall in it

In [ ]:
for i in range(0,8): # for rows
    for j in range(0,8): #for columns
        if i==0 or i ==7 or j ==0 or  j==7:
            print('*',end = '')
        else:
            print(' ', end ='')
    print()
        

********
*      *
*      *
*      *
*      *
*      *
*      *
********


***5x5 matrix and void in middle***

In [ ]:
for i in range(1,6):  #ROW
  for j in range(1,6): #COLUMN
    if j == 3 and i == 3 :
      print(" ",end="")
    else :  
      print("*",end="")
  print()  

*****
*****
** **
*****
*****


**5x5** **matrix**

In [ ]:
for i in range(5):
  for j in range(5):
    print("*",end="")
  print()  


*****
*****
*****
*****
*****


In [ ]:
for i in range(1,6) :   
  for j in range(1,i+1) :
    print(j,end="")
  print()      


1
12
123
1234
12345


In [ ]:
for i in range(1,6) :
  for j in range(1,6-i+1) :
    print(j,end="")
  print()  

12345
1234
123
12
1


In [ ]:
for i in range(1,8+1) :
  for j in range(1,8+1) :
    if i == 1 or i == 8 or j == 1 or j == 8 or i == 4 :
      print("*",end="")
    else :
      print(" ",end="")  
  print()  

********
*      *
*      *
********
*      *
*      *
*      *
********


PROBLEM 1) Rectangle

Sample Input: 4, 6

Sample Output:

123456

123456

123456

123456

Hint: 4 and 6 means 4 lines of numbers having 1..6 in each line

In [ ]:
row = int(input("Enter rows:"))
column = int(input("Enter column:"))

for i in range(row) :
  for j in range(1,column+1) :
    print(j,end="")
  print()  

Enter rows:9
Enter column:9
123456789
123456789
123456789
123456789
123456789
123456789
123456789
123456789
123456789


PROBLEM 2) Triangle - Left Justified. Draw a right-angled triangle of a given height.

Sample Input: 4

Sample Output:
#*
#**
#***
#****

In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for j in range(1,i+1) :
    print("*",end="")
  print()  

Enter row number:10
*
**
***
****
*****
******
*******
********
*********
**********


PROBLEM 3) Triangle - Left Justified. Draw a right-angled triangle of a given height.

Sample Input: 4

Sample Output :

1

12

123

1234


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for j in range(1,i+1) :
    print(j,end="")
  print()  

Enter row number:9
1
12
123
1234
12345
123456
1234567
12345678
123456789


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in range(1,row-i+1) :
    print(" ",end="")
  for star in range(1,i+1) :
    print("*",end="")
  print()    

Enter row number:10
         *
        **
       ***
      ****
     *****
    ******
   *******
  ********
 *********
**********


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in range(1,row-i+1) :
    print(" ",end="")
  for num in range(1,i+1) :
    print(num,end="")
  print()    

Enter row number:9
        1
       12
      123
     1234
    12345
   123456
  1234567
 12345678
123456789


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in range(1,row-i+1) :
    print(" ",end="")
  for star in range(1,i+1) :
    print("* ",end="")
  print()    

Enter row number:10
         * 
        * * 
       * * * 
      * * * * 
     * * * * * 
    * * * * * * 
   * * * * * * * 
  * * * * * * * * 
 * * * * * * * * * 
* * * * * * * * * * 


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in range(1,row-i+1) :
    print(" ",end="")
  for star in range(1,2*i) :
    print("*",end="")
  print()    

Enter row number:10
         *
        ***
       *****
      *******
     *********
    ***********
   *************
  ***************
 *****************
*******************


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in range(1,row-i+1) :
    print(" ",end="")
  for num in range(1,i*2) :
    print(num,end="")
  print()    

Enter row number:4
   1
  123
 12345
1234567


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in range(1,row-i+1) :
    print(" ",end="")
  for star in range(1,2*i) :
    print("*",end="")
  print()
for i in range(1,row) :
  for space in  range(1,i+1) :
    print(" ",end="")     
  for star in range(1,2*(row-i)) :
    print("*",end="")     
  print()               

Enter row number:10
         *
        ***
       *****
      *******
     *********
    ***********
   *************
  ***************
 *****************
*******************
 *****************
  ***************
   *************
    ***********
     *********
      *******
       *****
        ***
         *


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in  range(1,row-i+1) :
    print(" ",end="")
  for num in range(1,2*i) :
    print(num,end="")  
  print()
for i in range(1,row) :
  for space in range(1,i+1) :
    print(" ",end="")   
  for num in range(1,2*(row-i)) :
    print(num,end="") 
  print()  

Enter row number:4
   1
  123
 12345
1234567
 12345
  123
   1


In [ ]:
row = int(input("Enter row number:"))
col = int(input("Enter column number:"))
for i in range(1,row+1) :
  for j in range(1,col+1) :
    if i == 1 or i == row or j == 1 or j == col :
      print("*",end="")
    else :
      print(" ",end="")  
  print()    

Enter row number:6
Enter column number:8
********
*      *
*      *
*      *
*      *
********


In [ ]:
row = int(input("Enter row number:"))
col = int(input("Enter column number:"))
for i in range(1,row+1) :
  for j in range(1,col+1) :
    if i == 1 or i == row or j == 1 or j == col :
      print(j,end="")
    else :
      print(" ",end="")  
  print()    

Enter row number:6
Enter column number:8
12345678
1      8
1      8
1      8
1      8
12345678


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :

  for j in range(1,row+1) :

    if j == 1 or i == row or i == j :
      print("*",end="")
    else :
      print(" ",end="")  
  print()    

Enter row number:6
*     
**    
* *   
*  *  
*   * 
******


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for j in range(1,i+1) :
    if i == row or j == 1 or j == i :
      print("*",end="")
    else :
      print(" ",end="")  
  print() 

Enter row number:8
*
**
* *
*  *
*   *
*    *
*     *
********


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for j in range(1,i+1) :
    if i == row or j == 1 or j == i :
      print(j,end="")
    else :
      print(" ",end="")  
  print() 

Enter row number:8
1
12
1 3
1  4
1   5
1    6
1     7
12345678


In [ ]:
row = int(input("Enter row number:"))
for i in range(1,row+1) :
  for space in range(1,row-i+1) :
    print(" ",end="")
  for star in range(1,i+1) :
    if i == row or star == i or 